shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h

```diff
@@ -199,6 +206,7 @@ class FlutterPlatformViewLayerPool {
   /// This indicates that entries starting from 1 can be reused meanwhile the entry at position 0
   /// cannot be reused.
   size_t available_layer_index_ = 0;
+  mutable std::mutex layers_mutex_;
```

Comment:
Use `impeller::Mutex` from `//flutter/impeller/base/thread.h` and decorate `layers_` with IPLR_GUARDED_BY so you get static thread safety analysis.

Reply:
I'm curious, how can there be contention here?

Reply:
Hmm, I used to access this from multiple threads, but maybe I'm not anymore. Will take another pass through this.

Reply:
Removed